modules/git/pipeline/lfs_nogogit.go

```diff
@@ -116,6 +116,9 @@ func FindLFSFile(repo *git.Repository, hash git.SHA1) ([]*LFSResult, error) {
 				if err != nil {
 					return nil, err
 				}
+				if _, err := batchReader.Discard(1); err != nil {
```

Comment:
You have to remember not to forget this line. This may result in more bugs.

Reply:
Yeah. It's a fault of the current design - realistically we need to be abstracting this away - but our tests are catching this so it's actually ok. 

I'll have a think about it can be restructured.

Reply:
I've added a catch that will check for a missing discard and log it - but should work